File: e14/python/solution.py
class Solution:
    # @param nums: The integer array
    # @param target: Target number to find
    # @return the first position of target in nums, position start from 0 
    def binarySearch(self, nums, target):
        # write your code here
        if target < nums[0] or target > nums[-1]:
            return -1
        left = 0
        right = len(nums) - 1

        while right - left > 1:
            mid = (left + right) / 2
            if nums[mid] >= target:
                right = mid
            else: #nums[mid] < target:
                left = mid
        if nums[left] == target:
            return left
        elif nums[right] == target:
            return right
        else:
            return -1
        

File: e14/java/Solution.java
class Solution {
    /**
     * @param nums: The integer array.
     * @param target: Target to find.
     * @return: The first position of target. Position starts from 0.
     */
    public int binarySearch(int[] nums, int target) {
        //write your code here
        int size = nums.length;
        if (target < nums[0] || target > nums[size-1]) {
            return -1;
        }
        int left = 0;
        int right = size - 1;
        while (right -left > 1) {
            int mid = (left + right) / 2;
            if (nums[mid] >= target) {
                right = mid;
            }
            else {
                left = mid;
            }
        }
        if (nums[left] == target) {
            return (int)left;
        }
        else if (nums[right] == target) {
            return (int)right;
        }
        else {
            return -1;
        }
    }
}
